rust/lance/src/io/commit/dynamodb.rs

```diff
@@ -123,7 +126,11 @@ impl DynamoDBExternalManifestStore {
             .describe_table()
             .table_name(table_name)
             .send()
-            .await?;
+            .await
+            .map_err(|e| Error::IO {
+                message: format!("dynamodb error: {}", e,),
```

Comment:
can you keep the source error?

dynamodb formatter hides the read error if we only print `{}`

Reply:
```
349 |             .map_err(|e| Error::IO {
    |                       - binding `e` declared here
350 |                 message: format!("dynamodb error: {}, source: {:?}", e, e.into_source()),
    |                          -----------------------------------------------^---------------
    |                          |                                           |  |
    |                          |                                           |  move out of `e` occurs here
    |                          |                                           borrow of `e` occurs here
    |                          borrow later used here
```

did i do sth wrong here?

Reply:
We will have the `From<SDKError> for Error` back once this code is migrated to `lance-core`